beacon_node/execution_layer/src/engine_api/json_structures.rs

```diff
@@ -94,15 +94,18 @@ pub struct JsonExecutionPayload<T: EthSpec> {
     pub extra_data: VariableList<u8, T::MaxExtraDataBytes>,
     #[serde(with = "eth2_serde_utils::u256_hex_be")]
     pub base_fee_per_gas: Uint256,
-    #[superstruct(only(V3))]
-    // FIXME: can't easily make this an option because of custom deserialization..
-    #[serde(with = "eth2_serde_utils::u64_hex_be")]
-    pub excess_blobs: u64,
+    #[superstruct(only(V2))]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(default)]
+    #[serde(with = "eth2_serde_utils::u256_hex_be_opt")]
+    pub excess_data_gas: Option<Uint256>,
     pub block_hash: ExecutionBlockHash,
     #[serde(with = "ssz_types::serde_utils::list_of_hex_var_list")]
     pub transactions:
         VariableList<Transaction<T::MaxBytesPerTransaction>, T::MaxTransactionsPerPayload>,
-    #[superstruct(only(V2, V3))]
+    #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
sneaking in `#[serde(skip_serializing_if = "Option::is_none")]` and `#[serde(default)]` here because I think they'd be necessary? correct me if I'm wrong